src/fundamental/tpm-pcr.h

```diff
@@ -36,6 +36,7 @@ typedef enum UnifiedSection {
         UNIFIED_SECTION_DTB,
         UNIFIED_SECTION_PCRSIG,
         UNIFIED_SECTION_PCRPKEY,
+        UNIFIED_SECTION_SIGNED,
```

Comment:
Do we need a new section for this? What about always trying to boot with secure boot and falling back to trying to override the security policy? Although I guess there's nasty log messages when trying to boot a non-signed image with secure boot enabled?

Reply:
Also, this is causing compile time assertion failures in some of the TPM stuff that Lennart added recently.

Reply:
I wasn't thinking about a compile option, I was thinking about always trying secure boot first, and falling back to overriding the security policy after that if that fails. If we do that, we might not need the new .signed section at all.

Reply:
I guess that could be done. But it would incur a performance penalty as any unsigned image has to be loaded twice. And I have no idea what effect it would have on secure boot logs / TPM measurements.

Would you rather have the plague or cholera? :/

Reply:
Let's stick with the lnxsig section that Lennart suggested

Reply:
I'm not sure I follow how this is intended to work. Images can carry a signature, but it is also possible to store the hash of the image in DB. IOW, whether it can pass verification is not a property of the image per se.

Whether to use the override is really more of a per-system policy thing, so baking that into the image seems strange to me.

Reply:
> I'm not sure I follow how this is intended to work. Images can carry a signature, but it is also possible to store the hash of the image in DB. IOW, whether it can pass verification is not a property of the image per se.

I've changed it to .tlinux to convey that it doesn't have to be signed, just trusted.

> Whether to use the override is really more of a per-system policy thing, so baking that into the image seems strange to me.

Mmmh... maybe an EFI var to toggle this off makes more sense? (I personally don't have strong feelings about this)

@poettering Your thoughts?

Reply:
@ardbiesheuvel we are nesting two UEFI executables here in one PE file. The inner one (the actual kernel image) is included in the outer one (the "unified kernel image" UKI). The inner one is already covered by the outer's signature/hash validation, hence checking it again is in most ways really pointless. Now, people might still want to do that anyway (for example to get audit logs or so in order, or to avoid the vtable redirection hack), but whether to do that is really a decision of the *builder* of the UKI, since only the builder of the UKI knows if the inner image is signed by a good key or not.  By default we should assume that the inner kernels are not really signed by any key that the firmware knows (or even should know) – if it all.

And I doubt it makes much sense to validate the inner image via hashes at runtime, because it's not easy to even know the hashes to expect once the outer image is glued together (I mean, you can parse the PE file again, but who would do that...). After all users (unlike UKI builders) should never see the separate components the UKI is built from...